### PR TITLE
go-tools: pin go to 1.22

### PR DESCRIPTION
**Pull request by dakota-doordash**
## Description of changes

Switches go-tools to be compiled as a go 1.22 module. See https://github.com/dominikh/go-tools/issues/1500#issuecomment-1971702449.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/tools/go-tools/default.nix [hunk 1]
```diff
@@ -1,9 +1,9 @@
-{ buildGoModule
+{ buildGo122Module
```
Comment (kirillrdy) [inline pkgs/development/tools/go-tools/default.nix:1]: it is preferred way to override those in all-packages.nix

https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#changing-implicit-attribute-defaults

Comment (kirillrdy) [reply pkgs/development/tools/go-tools/default.nix:1]: also, congratulations on first contribution !

Comment (dakota-doordash) [reply pkgs/development/tools/go-tools/default.nix]: Thanks! Switched to the all-packages override method: https://github.com/NixOS/nixpkgs/pull/293629/commits/9b019e4dd465bb9f9e68371687fe66e4de9fe0bb.

**Comment by kirillrdy** [comment]
sorry for ad-hock feedback, can you update commit messages to match contribution guidelines

https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions

in this case, I would squash commits and have single

`go-tools: pin go to 1.22`

**Comment by dakota-doordash** [comment]
> 504b683

No problem. 
https://github.com/NixOS/nixpkgs/pull/293629/commits/504b68395d3af2ff00f22486d1e8f3155139e96b

**Comment by kirillrdy** [comment]
https://github.com/NixOS/nixpkgs/pull/290212 has been merged, but it will take some time before it reaches master

**Pull request merged**